contract/r/gnoswap/v1/router/utils.gno

```diff
@@ -165,3 +293,52 @@ func safeConvertToInt64(value *u256.Uint) int64 {
 	}
 	return int64(res)
 }
+
+// BuildSingleHopPath creates a single-hop route path string.
+// Format: "tokenA:tokenB:fee"
+//
+// Parameters:
+// - tokenA: input token address
+// - tokenB: output token address
+// - fee: pool fee (e.g., 500, 3000, 10000)
+//
+// Returns:
+// - string: formatted single-hop route path
+//
+// Example:
+//   - BuildSingleHopPath("gno.land/r/demo/wugnot", "gno.land/r/demo/usdc", 500)
+//     returns "gno.land/r/demo/wugnot:gno.land/r/demo/usdc:500"
+func BuildSingleHopRoutePath(tokenA, tokenB string, fee uint32) string {
+	if tokenA == "" || tokenB == "" {
+		panic("token addresses cannot be empty")
+	}
+	if tokenA == tokenB {
+		panic("tokenA and tokenB cannot be the same")
+	}
+
+	return ufmt.Sprintf("%s:%s:%s", tokenA, tokenB, formatUint(fee))
+}
+
+// BuildMultiHopPath creates a multi-hop route path string by connecting multiple single hops.
+// Format: "tokenA:tokenB:fee1*POOL*tokenB:tokenC:fee2*POOL*tokenC:tokenD:fee3"
+//
+// Parameters:
+// - hops: slice of single-hop path strings
+//
+// Returns:
+// - string: formatted multi-hop route path
+//
+// Example:
+//   - hops := []string{"tokenA:tokenB:500", "tokenB:tokenC:3000"}
+//     returns "tokenA:tokenB:500*POOL*tokenB:tokenC:3000"
+func BuildMultiHopRoutePath(hops ...string) string {
+	if len(hops) < 1 {
+		panic("multi-hop path requires at least 2 hops")
+	}
+
+	if len(hops) > 3 {
+		panic("multi-hop path supports maximum 3 hops")
+	}
+
+	return strings.Join(hops, POOL_SEPARATOR)
+}
```

Comment:
The `BuildMultiHopRoutePath` is not being used. It would be good to remove it. also, `BuildSingleHopRoutePath` is only used in tests, how about making it a test helper?